src/cueUtilities.js

```diff
@@ -319,60 +320,109 @@ module.exports = function (params, cy, api) {
+          return e.type.substring(0, 5) === 'mouse' || e.type === 'click';
+        }
+
+        function getTouchPageCoordinates(e) {
+          if (e.type === 'touchstart') {
+            return { x: event.touches[0].pageX,
+              y: event.touches[0].pageY
+            };
           }
-          if (nodeHasCue){
-            var expandcollapseRenderedStartX = node._private.data.expandcollapseRenderedStartX;
-            var expandcollapseRenderedStartY = node._private.data.expandcollapseRenderedStartY;
-            var expandcollapseRenderedRectSize = node._private.data.expandcollapseRenderedCueSize;
-            var expandcollapseRenderedEndX = expandcollapseRenderedStartX + expandcollapseRenderedRectSize;
-            var expandcollapseRenderedEndY = expandcollapseRenderedStartY + expandcollapseRenderedRectSize;
-                    
-                    var cyRenderedPos = event.renderedPosition || event.cyRenderedPosition;
-            var cyRenderedPosX = cyRenderedPos.x;
-            var cyRenderedPosY = cyRenderedPos.y;
-            var factor = (opts.expandCollapseCueSensitivity - 1) / 2;
-
-            if ( (Math.abs(oldMousePos.x - currMousePos.x) < 5 && Math.abs(oldMousePos.y - currMousePos.y) < 5)
-              && cyRenderedPosX >= expandcollapseRenderedStartX - expandcollapseRenderedRectSize * factor
-              && cyRenderedPosX <= expandcollapseRenderedEndX + expandcollapseRenderedRectSize * factor
-              && cyRenderedPosY >= expandcollapseRenderedStartY - expandcollapseRenderedRectSize * factor
-              && cyRenderedPosY <= expandcollapseRenderedEndY + expandcollapseRenderedRectSize * factor) {
-              if(opts.undoable && !ur)
-                ur = cy.undoRedo({
-                  defaultActions: false
-                });
-              if(api.isCollapsible(node))
-                if (opts.undoable){
-                  ur.do("collapse", {
-                    nodes: node,
-                    options: opts
-                  });
-                }
-                else
-                  api.collapse(node, opts);
-              else if(api.isExpandable(node))
-                if (opts.undoable)
-                  ur.do("expand", {
-                    nodes: node,
-                    options: opts
-                  });
-                else
-                  api.expand(node, opts);
-
-              // don't draw the cue if the group is expandable, as mouse won't be inside it
-              if (hoveredGroup && !isSelectedGroupsContains(hoveredGroup) && api.isExpandable(node)) {
-                hoveredGroup = null;
-              }
-              refreshCanvasImages();
+          return { x: event.changedTouches[0].pageX,
+            y: event.changedTouches[0].pageY
+          };
+        }
+
+        function isEventOnCueRegion(node, event) {
+          // needed so it matches the range for expandcollapses rendered locations
+          const TOUCH_OFFSET_Y = 43;
```

Comment:
How did you arrive at 43? If it's at all related to the styles or elements specified in our application, you can't have it as a constant in this unrelated library, it'll have to be passed in. Though I'm not sure why you need to use an offset to begin with.

Reply:
I'm guessing 43 is the position of the expand-collapse canvas in the browser window. You should be able to calculate this by looping up through the event target's parent elements and adding up the offsets.